cmake/modules/AddSwift.cmake

```diff
@@ -1069,7 +1066,8 @@ function(_add_swift_library_single target name)
   set(library_search_directories
       "${SWIFTLIB_DIR}/${SWIFTLIB_SINGLE_SUBDIR}"
       "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFTLIB_SINGLE_SUBDIR}"
-      "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_LIB_SUBDIR}")
+      "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_LIB_SUBDIR}"
+      "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_LIB_SUBDIR}/${SWIFTLIB_SINGLE_ARCHITECTURE}")
```

Comment:
We should not search both of these; either the arch is part of the path or it isn't.

Reply:
yeah, I can split that out. I originally didn't have two code paths for darwin/non-darwin code paths for adding libs but it ended up going that way because the logic for setting up cmake dependencies to the lipo task got a bit convoluted.

Reply:
Ah, that makes sense why you'd do it this way. I'd still like to go with the stricter way, though.

Reply:
fixed. I test if you are targeting "elf" and search in an arch specific directory. I could flip it to test on mach instead if the windows PE logic wants to do something similar to what I'm doing for ELF.

Reply:
Yeah, I suspect Mach-O fat binaries are the weird ones here, not the other way around.